src/main/java/bio/terra/grammar/google/BigQueryVisitor.java

```diff
@@ -41,6 +46,10 @@ public String visitColumn_expr(SQLParser.Column_exprContext ctx) {
     return String.format("`%s`.%s", alias, columnName);
   }
 
+  public static TableNameGenerator bqDatasetTableName(DatasetModel dataset) {
+    return tableName -> generateDatasetTableName(dataset, dataset.getName(), tableName);
+  }
+
   public static TableNameGenerator bqSnapshotTableName(SnapshotModel snapshot) {
     return tableName ->
         String.format("`%s.%s.%s`", snapshot.getDataProject(), snapshot.getName(), tableName);
```

Comment:
This seems to mirror `generateDatasetTableName`, did you think about abstracting that a little more, and reusing it here?